### PR TITLE
Fix cn_metrics.py not to fail on empty yaml input

### PR DESCRIPTION
**Pull request by beckerg**
Signed-off-by: Greg Becker <gbecker@micron.com>

### CODE REVIEW

**Comment by beckerg** [comment]
This hack should get SQA back on track, maybe some Python experts can help me out :-)

File: tools/scripts/cn_metrics.py [hunk 1]
```diff
@@ -12,7 +12,12 @@
 import json
 import pathlib
 import signal
-from typing import Any, Dict, OrderedDict, Optional
+from typing import Any, Optional
+
+try:
+    from typing import OrderedDict
+except ImportError:
+    from typing import Dict as OrderedDict
```
Comment (tristan957) [inline tools/scripts/cn_metrics.py:20]: Does the dict have to be ordered?

**Review by leroyjvargis — APPROVED** [review]

File: tools/scripts/cn_metrics.py [hunk 2]
```diff
@@ -69,6 +74,14 @@ def full_tree(ybuf: Optional[OrderedDict[str, Any]], opt):
     if ybuf["info"]["open"] == False:
         return
 
+    # TODO: How to test if ybuf[] contain at least one "nodes" stanza?
+    #
+    try:
+        for node in ybuf["nodes"]:
```
Comment (tristan957) [inline tools/scripts/cn_metrics.py:80]: `if "nodes" in ybuf`?

File: tools/scripts/cn_metrics.py (hunk 1 above)
Comment (beckerg) [reply tools/scripts/cn_metrics.py:20]: Only to preserve column order...

File: tools/scripts/cn_metrics.py (hunk 2 above)
Comment (beckerg) [reply tools/scripts/cn_metrics.py:80]: ybuf is some kind of monstrosity i don't grok, it returns NoneType if "nodes" is not in ybuf, but you shouldn't test explicitly for None/NoneType, not to mention the none stuff seems to vary depending on Python version, uhg...

Comment (tristan957) [reply tools/scripts/cn_metrics.py:80]: "nodes" in ybuf

tells you that there is a key called "nodes". If you are looking to make sure there is a node in the array, then do this:

`if ybuf.get("nodes", []): return`

**Pull request merged**